encoder/MysqlEncoder.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * @link https://github.com/tom--/dynamic-ar
+ * @copyright Copyright (c) 2015 Spinitron LLC
+ * @license http://opensource.org/licenses/ISC
+ */
+
+namespace spinitron\dynamicAr\encoder;
+
+use spinitron\dynamicAr\ValueExpression;
+
+class MysqlEncoder extends BaseEncoder
+{
+    /**
+     * Generate an SQL expression referring to the given dynamic column.
+     *
+     * @param string $name Attribute name
+     * @param string $type SQL datatype type
+     *
+     * @return string a Maria COLUMN_GET expression
+     */
+    public function columnExpression($name, $type = 'char')
+    {
+        $modelClass = $this->modelClass;
+        $sql = '[[' . $modelClass::dynamicColumn() . ']]';
+        foreach (explode('.', $name) as $column) {
+            $sql = "COLUMN_GET($sql, '$column' AS $type)";
+        }
+
+        return $sql;
+    }
+
+    /**
+     * Creates a dynamic column SQL expression representing the given attributes.
+     *
+     * @param array $attrs the dynamic attributes, which may be nested
+     *
+     * @return null|\yii\db\Expression
+     */
+    public function dynColExpression($attrs) {
+        if (!$attrs) {
+            return null;
+        }
+
+        $params = [];
+
+        // todo For now we only have Maria. Add PgSQL and generic JSON.
+        static::encodeArrayForMaria($attrs);
+        $sql = static::dynColSqlMaria($attrs, $params);
+
+        return new \yii\db\Expression($sql, $params);
+    }
```

Comment:
It is already there but as a thin wrapper to shorten the call. Would it make much difference?